src/rgw/rgw_reshard.cc

```diff
@@ -26,6 +26,37 @@ const string reshard_oid_prefix = "reshard.";
 const string reshard_lock_name = "reshard_process";
 const string bucket_instance_lock_name = "bucket_instance_lock";
 
+/* All primes up to 2000 used to attempt to make dynamic sharding use
+ * a prime numbers of shards. Note: this list also includes 1 for when
+ * 1 shard is the most appropriate, even though 1 is not prime.
+ */
+const std::set<uint16_t> RGWBucketReshard::reshard_primes = {
```

Comment:
if you make this an array, it can be stored in contiguous memory in the data segment, instead of having to do a lot of tiny heap allocations on startup. and if you use a std::initializer_list, you don't have to specify the array size in the header

then instead of the lower/upper_bound member functions of std::set, you can use the algorithms ie:
```c++
    auto it = std::upper_bound(reshard_primes.begin(), reshard_primes.end(),
                               requested_shards);
```

Reply:
Cool ideas! Implemented. Thanks!